client/sources/ok_test/models.py

```diff
@@ -100,9 +100,11 @@ def unlock(self, interact):
                                 for case in suite.cases])
         for suite_num, suite in enumerate(self.suites):
             for case_num, case in enumerate(suite.cases):
+                case_id = '{} > Suite {} > Case {}'.format(
```

Comment:
@ksteph @albert12132 - This is what I was trying to do: If a student messes with the suite number (by adding a case at the beginning or the end of an ok file), then any analytics scripts shouldn't break.

To that end, I was thinking that we should have the following as the identifier:

- assignment name
- The full text of the question (including anything that was locked- I'll explain below).
- The test name (e.g. 'Question 0')

So, let's say that we have a test case that looks like what Albert said:

```
>>> x = 5
>>> square(x)
25
>>> square(-x)
25
```

When a student first receives it and is on the first unlocking step, the test case actually looks like this:

```
>>> x = 5
>>> square(x)
hash(25)
>>> square(-x)
hash(25)
```

On the second unlocking step, the first thing is already unlocked, so the text is different and each prompt has a different identifier. So, the only way 2 things will have the same identifier is if in the same assignment, for the same question, the instructor decides to have the student do the exact same case twice for some reason. Also, the suite number doesn't matter so reordering tests and the like doesn't change identifiers.

Hashes of things are easier for me to think about, which is why I'm biased. Now that I think about it, hashing is completely unnecessary here. XD